azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -295,23 +295,23 @@ def run(self, goal_state):
             requested_version = self.__get_requested_version(agent_family)
             agent_manifest = None  # This is to make sure fetch agent manifest once per update
             warn_msg = ""
-            if requested_version is None:
+            if requested_version is not None and agent_family.is_version_from_rsm:
+                self._is_requested_version_update = True
+                # Save the requested version to report back
+                GAUpdateReportState.report_expected_version = requested_version
+                # Remove the missing requested version warning once requested version becomes available
+                if "Missing requested version" in GAUpdateReportState.report_error_msg:
+                    GAUpdateReportState.report_error_msg = ""
+            else:
                 # Do not proceed with update if self-update needs to download the manifest again with in an hour
                 if not self.__should_agent_attempt_manifest_download():
                     return
-                if conf.get_enable_ga_versioning():  # log the warning only when ga versioning is enabled
+                if conf.get_enable_ga_versioning() and not agent_family.is_version_from_rsm:  # log the warning only when ga versioning is enabled
```

Comment:
but aren't you planning on enabling ga versioning? (and hence this warning would always be there until RSM updates are enabled?)

Reply:
yet, I'll do it in separate PR. We only log the warning when RSM updates are enabled and still missing requested version